app/models/comment.rb

```diff
@@ -163,6 +164,37 @@ def root_exists?
     ancestry && Comment.exists?(id: ancestry)
   end
 
+  def edited(new_attributes, by:)
+    self.class.transaction do
+      old_attributes = attributes
+      updated = update(new_attributes)
+
+      if updated
+        changes = new_attributes.to_h.each_with_object({}) do |(attribute, new_value), hash|
```

Comment:
I was hoping we could use something like the following but alas a comment update results in 4 distinct updates the last one only updates `updated_at`.

```ruby
comment.saved_changes.except("updated_at")
```